src/Compilers/Core/Portable/CodeGen/SequencePointList.cs

```diff
@@ -124,6 +124,35 @@ public ImmutableArray<Cci.SequencePoint> GetSequencePoints(DebugDocumentProvider
                 current = current._next;
             }
 
+            // Find the document for the first non-hidden sequence point (issue #4370)
+            current = this;
+            while (current != null)
```

Comment:
I'd extract this loop to a method. It's self contained and will be easier to exit the outer loop.